### ISSUE TITLE
Problem to Enable an Audit

### ISSUE THREAD

**Issue by leodlsrt** [abedra/libvault#71]
Hi, 
I've a problem when i want to enable an audit of type "file", i don't know if it comes from my code or from the library...

When I trying via CURL, this works
![image](https://user-images.githubusercontent.com/50668162/103417550-5eec1680-4b8b-11eb-94da-59840b6bd6f5.png)

Result in Vault Server Console : 
![image](https://user-images.githubusercontent.com/50668162/103417563-6c090580-4b8b-11eb-9636-82f879d50a9a.png)

But when I try this with the C++ library, it gives me a 400 Bad Request "enable audit mount failed: path=file/ error="unknown backend type: """

This is my code : 
![image](https://user-images.githubusercontent.com/50668162/103417479-23514c80-4b8b-11eb-8424-8c04cbd82fab.png)

Result in C++ Console : 
![image](https://user-images.githubusercontent.com/50668162/103417684-ff423b00-4b8b-11eb-902b-b78c7537438e.png)

Result in Vault Server Console : 
![image](https://user-images.githubusercontent.com/50668162/103417578-7f1bd580-4b8b-11eb-9065-d413861dafd9.png)

Thanks for you response

**Comment by abedra** [comment]
I just took a look and found two unique issues that cause this. The first is a simple bug where the variant of `HttpConsumer::put` being used actually uses the `POST` method. That's evident in the console output above and was an error in the initial implementation. The next, and larger issue, is in part due to the way your example is constructing `Vault::Parameters`, and in total a missing part of the library. `Vault::Parameters` is specialized to `std::unordered_map<std::string, std::string>`, which doesn't support complex parameters. Since this particular Vault endpoint requires complex parameters, it will need to be supported. 

After this is fixed, your code should follow this pattern for parameter construction

```cpp
Vault::Parameters params {
  {"type", "file"},
  {"options", {
    {"file_path", "log"}
  }
};
```

The reason for your error was that you were constructing a map with a key name the full JSON string literal and a value of an empty string. 

In this particular case, I'll add a direct fix for a complex parameter type and avoid any major breaking changes to the library, but upon looking, it will probably be a good idea to leave the full power of the map template in the `Parameters` alias. This will be a pretty major breaking change and force a fair amount of refactoring, so I'll save that for another changeset.

**Comment by abedra** [comment]
This has been fixed in release `0.36.0`. A full working example is at https://github.com/abedra/libvault/blob/master/example/administration/audit/example.cpp

**Comment by leodlsrt** [comment]
Hi Aaron,
Thanks for the quick fix, however, I see the same problem to enable a Secret Engine (mounts), when I try to do like this, the code doesn't execute it doesn't make any request to the server, it runs in a loop...

I think I should just add an optional parameter for the options on this line :
https://github.com/abedra/libvault/blob/a84359152c8e83d26e90f1d919572677e59b614f/include/VaultClient.h#L645

If needed, here is my code :
![image](https://user-images.githubusercontent.com/50668162/104162267-91500c00-53f4-11eb-88a2-080b02135d02.png)

Thank you very much,
Yours sincerely,
leodlsrt
